setup.py

```diff
@@ -1,8 +1,62 @@
-from distutils.core import setup
-from Cython.Build import cythonize
-import numpy
-
-setup(
-    ext_modules=cythonize("*.pyx", language="c++"),
-    include_dirs=[numpy.get_include()]
-)
+"""
+Implements the MDLP discretization criterion from Usama Fayyad's paper
+"Multi-Interval Discretization of Continuous-Valued Attributes for
+Classification Learning."
+"""
+
+from setuptools import Extension, find_packages, setup
+
+if __name__ == '__main__':
+    # see https://stackoverflow.com/a/42163080 for the approach to pushing
+    # numpy and cython dependencies into extension building only
+    try:
+        # if Cython is available, we will rebuild from the pyx file directly
+        from Cython.Distutils import build_ext
+        sources = ['mdlp/_mdlp.pyx']
+    except:
+        # else we build from the cpp file included in the distribution
+        from setuptools.command.build_ext import build_ext
+        sources = ['mdlp/_mdlp.cpp']
+
+    class CustomBuildExt(build_ext):
+        """
```

Comment:
Use PEP 257 here please.
https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings

Reply:
I rewrote this docstring specifically in response to your previous request and link to PEP 257 and I believe this docstring follows that standard. Can you be more specific about what you would like to be changed?

Reply:
I meant something like this, sorry if it wasn't clear:

```
"""Custom build_ext class to defer numpy imports until needed.

Overrides the run command for building an extension and adds in numpy
include dirs to the extension build. Doing this at extension build time
allows us to avoid requiring that numpy be pre-installed before
executing this setup script.
"""
```

Reply:
sure.

Reply:
To be clear. I understand that you are just asking that the docstring start on the same line as the opening quotes, right? Of course I should follow the style of the package and do this but the PEP does not insist on this FYI:

"The summary line may be on the same line as the opening quotes or on the next line."

Reply:
Hmm, interesting.